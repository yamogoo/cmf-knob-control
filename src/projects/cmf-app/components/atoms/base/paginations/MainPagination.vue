<template lang="pug">
BasePagination(
    v-slot="{ isActive }"
    :items-count="itemsCount"
    :sid="sid"
)
    div(
        ref="refItem"
        :class="[`${APP_PREFIX}-pagination__item-shape`, isActive ? Classes.ACTIVE : Classes.NORMAL]"
        :style="{width: size + 'px', height: size + 'px', borderRadius: size/2 + 'px'}"
    )
</template>

<script setup lang="ts">
import { APP_PREFIX } from '@app/config';
import BasePagination, { type Props } from './BasePagination.vue';

interface BaseProps extends Props {
    size?: number
}

withDefaults(defineProps<BaseProps>(), {
    sid: 0,
    size: 8
});

</script>

<script lang="ts">
enum Classes {
    ACTIVE = 'active',
    NORMAL = 'normal'
}
</script>

<style lang="scss">
.#{$APP_PREFIX}-pagination {
    position: relative;
    display: flex;
    @include box(auto, 100%);
    gap: 6px;

    &__item {

        &-shape {
            &.normal {
                background-color: darken($GRAY, 26%);
            }

            &.active {
                background-color: $CARDAMON_LIGHT;
            }
        }
    }
}
</style>