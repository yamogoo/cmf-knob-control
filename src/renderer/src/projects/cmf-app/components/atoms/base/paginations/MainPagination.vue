<template lang="pug">
BasePagination(
    v-slot="{ isActive }"
    :items-count="itemsCount"
    :sid="sid"
)
    div(
        ref="refItem"
        :class="[`${APP_PREFIX}-pagination__item-indicator`, isActive ? Classes.ACTIVE : Classes.NORMAL]"
        :style="{width: size + 'px', height: size + 'px', borderRadius: size/2 + 'px'}"
        data-testid="pagination__item-indicator"
    )
</template>

<script setup lang="ts">
import { APP_PREFIX } from '@app/config'
import BasePagination, { type Props } from './BasePagination.vue'

interface BaseProps extends Props {
  sid: number
  size?: number
}

withDefaults(defineProps<BaseProps>(), {
  sid: 0,
  size: 8
})
</script>

<script lang="ts">
enum Classes {
  ACTIVE = 'active',
  NORMAL = 'normal'
}
</script>

<style lang="scss">
.#{$APP_PREFIX}-pagination {
  position: relative;
  display: flex;
  @include box(auto, 100%);
  gap: 6px;

  &__item {
    &-indicator {
      &.normal {
        background-color: darken($GRAY, 26%);
      }

      &.active {
        background-color: $CARDAMON_LIGHT;
      }
    }
  }
}
</style>
