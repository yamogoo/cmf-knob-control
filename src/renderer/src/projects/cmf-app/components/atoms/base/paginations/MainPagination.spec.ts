import { describe, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'

import MainPagination from './MainPagination.vue'

describe('MainPagination', () => {
  test.each([[6, 0]])('should have indicators count (%i)', (itemsCount, sid) => {
    const component = mount(MainPagination, {
      props: { sid, itemsCount }
    })

    const indicatorEls = component.findAll('[data-test-id="pagination__item-indicator"]')
    const length = indicatorEls.length

    expect(length).toBe(itemsCount)
    expect(length).toMatchSnapshot()
  })

  test.each([12])('an indicator should have size (%i)', (size) => {
    const component = mount(MainPagination, {
      props: {
        size,
        sid: 1,
        itemsCount: 6
      }
    })

    const indicatorEls = component.findAll('[data-test-id="pagination__item-indicator"]')
    const indicatorStyle = indicatorEls[0].attributes('style')

    expect(indicatorStyle).toContain(`width: ${size}px;`)
    expect(indicatorStyle).toContain(`height: ${size}px;`)
    expect(indicatorStyle).toContain(`border-radius: ${size / 2}px;`)
    expect(indicatorStyle).toMatchSnapshot()
  })

  test.each([[0, 6]])('first indicator should have "active" class', (sid, itemsCount) => {
    const component = mount(MainPagination, {
      props: {
        sid,
        itemsCount
      }
    })

    const indicatorEls = component.findAll('[data-test-id="pagination__item-indicator"]')
    const firstIndicator = indicatorEls[0]
    const isActiveExist = firstIndicator.classes('active')

    expect(isActiveExist).toBe(true)
    expect(isActiveExist).toMatchSnapshot()
  })
})
