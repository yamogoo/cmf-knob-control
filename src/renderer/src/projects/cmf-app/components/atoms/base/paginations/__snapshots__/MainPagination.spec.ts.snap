// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainPagination > an indicator should have size (12) 1`] = `"width: 12px; height: 12px; border-radius: 6px;"`;

exports[`MainPagination > first indicator should have "active" class 1`] = `true`;

exports[`MainPagination > should have indicators count (6) 1`] = `6`;
